src/test/scala/chisel3/internal/NameCollisionSpec.scala

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.stage.ChiselStage
+
+class NameCollisionSpec extends ChiselFlatSpec with Utils {
+  behavior.of("Builder")
+
+  it should "error on duplicated names with a correct message" in {
+    (the[ChiselException] thrownBy extractCause[ChiselException] {
+      (new ChiselStage).emitChirrtl(
```

Comment:
```suggestion
      (new ChiselStage).elaborate(
```
These error so no need for chirrtl, can just elaborate. Also the method on `circt.stage.ChiselStage` is `emitCHIRRTL` for some reason